src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/DelegatedCompletionListProvider.cs

```diff
@@ -188,7 +158,7 @@ private static VSInternalCompletionContext RewriteContext(VSInternalCompletionCo
                 return null;
             }
 
-            var previousCharacterProjection = GetProjection(projection.AbsoluteIndex - 1, codeDocument, sourceText);
+            var previousCharacterProjection = await documentContext.GetProjectionAsync(projection.AbsoluteIndex - 1, _documentMappingService, cancellationToken).ConfigureAwait(false);
```

Comment:
AHhh I likeee, how about we have the document context factory pass in a mapping service?